client/Panels/Topics/Model.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { useState, useEffect } from 'react';
+import { GET_TOPICS } from 'Queries/Topics';
+import { useQuery } from '@apollo/client';
+import {
+  useTopicsModelType,
+  useGetTopicsResultType,
+} from './useTopicsModel.types';
+
+export const useTopicsModel = (): useTopicsModelType => {
+  const useGetTopics = (filter?: string): useGetTopicsResultType => {
+    // Use debounce to ensure char-by-char changes to the filter string do not cause floods of useQuery network calls
+    const debouncedFilter = useDebounce(filter, 500);
+    // Caching in useQuery means that new network calls are only made when the debouncedFilter value changes
+    return useQuery(GET_TOPICS, { variables: { filter: debouncedFilter } });
+  };
+
+  return {
+    useGetTopics,
```

Comment:
This is interesting (and not what I expected - not to say that is wrong!). I expected that we would also be owning/exporting the state for the filter - ie its value and function to update it, which in fact is a (lodash) debounced version. When that debounce clears, then  the value is set, firing the query. Does this mean the view is making use of a `useState` call to track the filter value?

Edit: having seen the code below, I see this is the case. My preference would be the Model does all hook work, and that the view imports as few hooks as possible (for easy sharing of the business logic/keeping the view layers thin)